sdk/src/Core/Amazon.Runtime/Credentials/Internal/_bcl45+netstandard/SsoTokenUtils.cs

```diff
@@ -99,13 +99,17 @@ public static string AsJson(this SsoToken token)
 
         /// <summary>
         /// This determines whether the <seealso cref="SsoToken.RegistrationExpiresAt"/> field is 5 minutes within expiration
+        /// Existing cached tokens, or tokens written out by other systems may contain null values for the RegistrationExpiresAt field.
+        /// In these cases we treat the client registration as expired and generate a new token.
         /// </summary>
         /// <param name="token">The sso token</param>
         /// <returns>This returns true if <seealso cref="SsoToken.RegistrationExpiresAt"/> is within 5 minutes of expiration. False otherwise</returns>
         public static bool RegisteredClientExpired(this SsoToken token)
         {
             if (null == token)
                 throw new ArgumentNullException(nameof(token));
+            if (string.IsNullOrEmpty(token.RegistrationExpiresAt))
+                    return true;
```

Comment:
I haven't actually used this, does the user experience differ for whether we're calling `GenerateNewTokenAsync` vs. `RefreshTokenAsync`?

Still debating the two proposals from @awschristou in https://github.com/aws/aws-sdk-net/pull/3083/files#r1389714680.

If using the SDK with a tool that never sets `RegistrationExpiresAt` originally, will we be forcing new tokens to eagerly even if the registration had a lot of time left?

Reply:
I think that knowing the system is working with clean data moving forward is better than the potential risk of leaving users in a scenario where they are unable to establish a connection, without manual intervention (manually deleting the cached tokens).